twenty-first/src/shared_math/b_field_element.rs

```diff
@@ -552,286 +552,251 @@ mod b_prime_field_element_test {
         }
     }
 
-    #[should_panic(
-        expected = "Byte representation must represent a valid B field element, less than the quotient."
-    )]
-    #[test]
-    fn disallow_conversion_of_u8_array_outside_range() {
-        let bad_bfe_array: [u8; 8] = [u8::MAX; 8];
-        println!("bad_bfe_array = {bad_bfe_array:?}");
-        let _value: BFieldElement = bad_bfe_array.into();
+    proptest! {
+        #[test]
+        fn addition_is_associative(
+            element_0: BFieldElement,
+            element_1: BFieldElement,
+            element_2: BFieldElement
+        ) {
+            prop_assert_eq!(
+                (element_0 + element_1) + element_2,
+                element_0 + (element_1 + element_2)
+            );
+        }
     }
 
-    #[test]
-    fn simple_value_test() {
-        let zero: BFieldElement = BFieldElement::new(0);
-        assert_eq!(0, zero.value());
-        let one: BFieldElement = BFieldElement::one();
-        assert_eq!(1, one.value());
+    proptest! {
+        #[test]
+        fn multiplication_is_associative(
+            element_0: BFieldElement,
+            element_1: BFieldElement,
+            element_2: BFieldElement
+        ) {
+            prop_assert_eq!(
+                (element_0 * element_1) * element_2,
+                element_0 * (element_1 * element_2)
+            );
+        }
+    }
+
+    proptest! {
+        #[test]
+        fn multiplication_distributes_over_addition(
+            element_0: BFieldElement,
+            element_1: BFieldElement,
+            element_2: BFieldElement,
+        ) {
+            prop_assert_eq!(
+                element_0 * (element_1 + element_2),
+                element_0 * element_1 + element_0 * element_2
+            );
+        }
+    }
 
-        let neinneinnein = BFieldElement::new(999);
-        assert_eq!(999, neinneinnein.value());
+    proptest! {
+        #[test]
+        fn multiplication_with_inverse_gives_identity(bfe in arbitrary_non_zero_bfield_element()) {
+            prop_assert!((bfe.inverse() * bfe).is_one());
+        }
     }
 
-    #[test]
-    fn simple_generator_test() {
-        assert_eq!(BFieldElement::new(7), BFieldElement::generator());
-        assert!(BFieldElement::new(7)
-            .mod_pow(((1u128 << 64) - (1u128 << 32)) as u64)
-            .is_one());
-        assert!(!BFieldElement::new(7)
-            .mod_pow(((1u128 << 64) - (1u128 << 32)) as u64 / 2)
-            .is_one());
+    proptest! {
+        #[test]
+        fn division_by_self_gives_identity(bfe in arbitrary_non_zero_bfield_element()) {
+            prop_assert!((bfe / bfe).is_one());
+        }
+    }
+
+    proptest! {
+        #[test]
+        fn values_larger_than_modulus_are_handled_correctly(large_value in BFieldElement::P..) {
+            let bfe = BFieldElement::new(large_value);
+            let expected_value = large_value - BFieldElement::P;
+            prop_assert_eq!(expected_value, bfe.value());
```

Comment:
Wouldn't this test be marginally more readable as a one-liner? 

```rust
prop_assert_eq!(large_value - BFieldElement::P, BFieldElement::new(large_value).value());
```

Reply:
I disagree with this one: the suggested one-liner is (a) more difficult to digest and (b) exceeds the line length limit.